datastore/importers/nmdb-import-arista-show-lldp-neighbor/Parser.cpp

```diff
@@ -0,0 +1,340 @@
+        )]
+    ;
+
+  restOfLine =
+    *(qi::char_ - qi::eol) 
+    >> qi::eol
+    ;
+
+  inQuotes =
+    '"'
+    > *(qi::char_ - '"')
+    > '"'
+    ;
+
+  token =
+    +qi::ascii::graph
+    ;
+
+  csvToken =
+    +(qi::ascii::graph - qi::char_(','))
+    ;
+
+  ignoredLine =
+      (+token > qi::eol)
+    | (qi::eol)
+    ;
+
+  BOOST_SPIRIT_DEBUG_NODES(
+      //(start)
+      (noDetailConfig)
+        (noDetailTableInfo)
+        (noDetailHeader)
+        (noDetailEntry)
+      (detailConfig)
+        (detailHeader)
+        (detailEntry)
+          (detailNeighborLine)
+          (detailDiscovered)
+          (detailChassisId)
+          (detailPortId)
+          (detailPortDescription)
+          (detailSystemName)
+          (detailSystemDescription)
+          (detailCapabilities)
+      (port)
+      //(restOfLine)
+      (inQuotes)
+      //(ignoredLine)
+      //(token)
+      //(csvToken)
+    );
+}
+
+// =============================================================================
+// Parser helper methods
+// =============================================================================
+std::string
+Parser::getDevice(const std::string& hostname)
+{
+  return hostname.substr(0, hostname.find("."));
+}
+
+// currently disabled
+// void
+// Parser::updateIpAddrs()
+// {
+//   for (auto ip : nd.ipAddrs) {
+//     ip.addAlias(nd.curHostname, "lldp import");
+//     d.ipAddrs.push_back(ip);
+//   }
+// }
+
+void
+Parser::addVlan(uint16_t vlanId)
+{
+  nd.curVlans.emplace_back(vlanId);
+}
+
+void
+Parser::updateInterfaces()
+{
+  nmdo::InterfaceNetwork iface {nd.curIfaceName};
+  iface.setPartial(true);
+  iface.setState(true);
+  iface.setDiscoveryProtocol(true);
+  if (!nd.curPortDescription.empty()) {
+    iface.setDescription(nd.curPortDescription);
+  }
+
+  if (!nd.curMacAddr.empty()) {
+    nmdo::MacAddress macAddr {nd.curMacAddr};
+    iface.setMacAddress(macAddr);
+  }
+
+  if (!nd.curVlans.empty()) {
+    for (auto& vlanId : nd.curVlans) {
+      iface.addVlan(vlanId);
+    }
+  }
+
+  d.interfaces.emplace_back(iface, getDevice(nd.curHostname));
+}
+
+void
+Parser::updatePhysicalConnection()
+{
+  nmdo::PhysicalConnection physCon;
+  physCon.setSrcIfaceName(nd.srcIfaceName);
+  physCon.setDstDeviceId(getDevice(nd.curHostname));
+  physCon.setDstIfaceName(nd.curIfaceName);
+
+  d.physConnections.emplace_back(physCon);
+}
+
+void
+Parser::updateDeviceInformation()
+{
+  nmdo::DeviceInformation devInfo;
+
+  devInfo.setDeviceId(getDevice(nd.curHostname));
+  if (!nd.curDeviceType.empty()) {
+    devInfo.setDeviceType(nd.curDeviceType);
+  }
+  if (!nd.curSysDescription.empty()) {
+    devInfo.setDescription(nd.curSysDescription);
+  }
+
+  d.devInfos.push_back(devInfo);
+}
+
+void
+Parser::finalizeData()
+{
+  if (!nd.curHostname.empty())  {
+    // normalize hostname
+    auto temp {nd.curHostname};
+    auto pos = temp.find('(');
+    if (std::string::npos != pos) {
+      temp = temp.substr(0, pos);
+    }
```

Comment:
I don't think you need to do this.  LLDP does not do the same hostname format that CDP does.